pycognito/__init__.py

```diff
@@ -889,3 +895,106 @@ def admin_update_user_pool_client(self, pool_id: str, client_id: str, **kwargs):
             ClientId=client_id,
             **kwargs,
         )
+
+    def associate_software_token(self):
+        """
+        Get the SecretCode used for Software Token. SecretCode use to set up Software Token MFA.
+        :return: SecretCode
+        :rtype: string
+        """
+        response = self.client.associate_software_token(AccessToken=self.access_token)
+        return response["SecretCode"]
+
+    def verify_software_token(self, code, device_name):
+        """
+        Verify the value generated by TOTP to complete the registration of Software Token MFA.
+        :param code: The value generated by TOTP
+        :param device_name: Device name to register (optional)
+        :return: verify success
+        :rtype: bool
+        """
+        response = self.client.verify_software_token(
+            AccessToken=self.access_token, UserCode=code, FriendlyDeviceName=device_name
+        )
+        return response["Status"] == "SUCCESS"
+
+    def set_user_mfa_preference(self, sms_mfa, software_token_mfa, preferred=None):
+        """
+        Register the preference of MFA.
+        :param sms_mfa: Enable SMS MFA.
+        :type sms_mfa: bool
+        :param software_token_mfa: Enable Software Token MFA.
+        :type software_token_mfa: bool
+        :param preferred: Which is the priority, SMS or Software Token? The expected value is "SMS" or "SOFTWARE_TOKEN". However, it is not needed only if both of the previous arguments are False.
+        :type preferred: string
+        :return:
+        """
+        sms_mfa_settings = {"Enabled": bool(sms_mfa), "PreferredMfa": False}
+        software_token_mfa_settings = {
+            "Enabled": bool(software_token_mfa),
+            "PreferredMfa": False,
+        }
+        if not (bool(sms_mfa) or bool(software_token_mfa)):
+            # Disable MFA
+            pass
+        if preferred == "SMS":
+            sms_mfa_settings["PreferredMfa"] = True
+        elif preferred == "SOFTWARE_TOKEN":
+            software_token_mfa_settings["PreferredMfa"] = True
+        else:
+            raise ValueError(
+                "preferred is not the correct value.\nThe expected value is SMS or SOFTWARE_TOKEN."
+            )
+        self.client.set_user_mfa_preference(
+            SMSMfaSettings=sms_mfa_settings,
+            SoftwareTokenMfaSettings=software_token_mfa_settings,
+            AccessToken=self.access_token,
+        )
+
+    def respond_to_software_token_mfa_challenge(self, code, mfa_tokens=None):
+        """
+        Respons challenge to software token of MFA.
+        :param code: software token MFA code.
+        :type code: string
+        :param code: mfa_token stored in MFAChallengeException. Not required if you have not regenerated the Cognito instance.
+        :type code: string
```

Comment:
The parameter name is `mfa_tokens`, but the doc calls it `code`. And it's some collection, not a string, right?